go-controller/pkg/config/config.go

```diff
@@ -406,32 +456,30 @@ func InitConfig(ctx *cli.Context, defaults *Defaults) error {
 		}
 		logrus.Infof("Parsed config file %s", f.Name())
 		logrus.Infof("Parsed config: %+v", cfg)
-
-		if defaults != nil {
-			fillDefaults(&cfg, defaults)
-		}
 	}
 
-	// command-line > config file > default values
-	Default.MTU = getConfigInt(cliConfig.Default.MTU, cfg.Default.MTU, 1400)
-	Default.ConntrackZone = getConfigInt(cliConfig.Default.ConntrackZone, cfg.Default.ConntrackZone, 64000)
-
-	Logging.File = getConfigStr(cliConfig.Logging.File, cfg.Logging.File, "/var/log/ovn-kubernetes.log")
-	Logging.Level = getConfigInt(cliConfig.Logging.Level, cfg.Logging.Level, 4)
+	if defaults == nil {
+		defaults = &Defaults{}
+	}
 
-	CNI.ConfDir = getConfigStr(cliConfig.CNI.ConfDir, cfg.CNI.ConfDir, "/etc/cni/net.d")
-	CNI.Plugin = getConfigStr(cliConfig.CNI.Plugin, cfg.CNI.Plugin, "ovn-k8s-cni-overlay")
+	// Build config that needs no special processing
+	overrideFields(&Default, &cfg.Default)
+	overrideFields(&Default, &cliConfig.Default)
+	overrideFields(&Logging, &cfg.Logging)
+	overrideFields(&Logging, &cliConfig.Logging)
```

Comment:
I don't see where Logging.Level is being honored. In my tests it doesn't matter to which value I change it.
We should follow with a call to logrus.SetLevel for that.
Something like this:
```
func setLogLevel(level int) error {
    switch level {
    case 5:
        logrus.SetLevel(logrus.DebugLevel)
        return nil
    case 4:
        logrus.SetLevel(logrus.InfoLevel)
        return nil
    case 3:
        logrus.SetLevel(logrus.WarnLevel)
        return nil
    case 2:
        logrus.SetLevel(logrus.ErrorLevel)
        return nil
    case 1:
        logrus.SetLevel(logrus.FatalLevel)
        return nil
    case 0:
        logrus.SetLevel(logrus.PanicLevel)
        return nil
    default:
        return fmt.Errorf("setLogLevel: given log level is out of range")
    }
}
```

Reply:
@alinbalutoiu it should get called by the individual app, eg ovnkube.go:

```
func runOvnKube(ctx *cli.Context) error {
        if err := config.InitConfig(ctx, nil); err != nil {
                return err
        }

        // Process log flags
        logrus.SetLevel(logrus.Level(config.Logging.Level))
        logrus.SetOutput(os.Stderr)
```

Reply:
Ah, didn't check ovnkube, this happened to me when using the CNI plugin.
eg ovn-k8s-cni-plugin.go: https://github.com/openvswitch/ovn-kubernetes/blob/master/go-controller/cmd/ovn-k8s-cni-overlay/ovn-k8s-cni-overlay.go#L261-L272

Wouldn't it make more sense to do it right after we read the value? This for example also gets ignored since it's debug log and the level will be set only after the function returns. https://github.com/openvswitch/ovn-kubernetes/blob/master/go-controller/pkg/config/config.go#L438-L443